chainservice/builder.go

```diff
@@ -240,13 +240,11 @@ func (builder *Builder) buildBlockDAO(forTest bool) error {
 
 	var indexers []blockdao.BlockIndexer
 	indexers = append(indexers, builder.cs.factory)
-	if !builder.cfg.Chain.EnableAsyncIndexWrite {
-		if builder.cs.bfIndexer != nil {
-			indexers = append(indexers, builder.cs.bfIndexer)
-		}
-		if builder.cs.indexer != nil {
-			indexers = append(indexers, builder.cs.indexer)
-		}
+	if !builder.cfg.Chain.EnableAsyncIndexWrite && builder.cs.indexer != nil {
+		indexers = append(indexers, builder.cs.indexer)
+	}
+	if builder.cs.bfIndexer != nil {
+		indexers = append(indexers, builder.cs.bfIndexer)
```

Comment:
`bfIndexer` should always be added

Reply:
This is definitely a bug, which won't work if `EnableAsyncIndexWrite` is true. After this PR, the logic is identical to previous implementation. But why `bfIndexer` should be updated synchronically, instead of something like https://github.com/iotexproject/iotex-core/blob/5237ffb5a3b93d9dde73595c4f28363df24c43a3/chainservice/builder.go#L354?

Reply:
right, the PR is to make logic identical as before
checked the `bfIndexer` code, it is to save all logs in every block, so I think it can also be wrapped into a `indexBuilder` and handled the similar way by `AddSubscriber`, can be another PR